Add glossary term 'escaping text' (#37444)
mdn/content: files/en-us/glossary/escaping_text/index.md

---

> title: Escaping Text

Author: ```suggestion
title: Escaping Character
```

> slug: Glossary/Escaping_Text

Author: ```suggestion
slug: Glossary/Escaping_Character
```

---

> title: Escaping Text

Reviewer: A glossary title should be a concrete term, not a concept. I recommend either "escape character" or "escape sequence.

Author: same idea, though not sure which one to choose, so i pick the one the issue title suggested

@estelle what's your opinion on this name?

Reviewer 2: i think "escape character" will be searched for more. ("escaping text" would be a good title for a guide).

slug: Glossary/Escaping_Text
page-type: glossary-definition
---

{{GlossarySidebar}}

> An **escape character** is a {{glossary("character")}} that invokes an alternative interpretation on the following character in a sequence. The escape character is a character that signals that the following character should be interpreted differently.

Reviewer 2: ```suggestion
An **escape character** is a {{glossary("character")}} that invokes an alternative interpretation of the following character in a sequence, signaling that the following character should be interpreted differently.
```

> An **escape character** is a {{glossary("character")}} that invokes an alternative interpretation of the following character in a sequence, signaling that the following character should be interpreted differently.
> 
> In many programming languages, an escape character also forms some **escape sequences** that has a special meaning rather than the literal characters, marked by one or more preceding characters and possibly terminating characters.

Reviewer: ```suggestion
An **escape character** is a {{glossary("character")}} that causes one or more characters that follow it to be interpreted differently. This forms an **escape sequence**, which is often used to represent a character that has an alternative meaning when printed literally, such as the quote character in a string literal. Escape sequences can have other usage too, especially in [regular expressions](/en-US/docs/Web/JavaScript/Reference/Regular_expressions#escape_sequences).
```

Author: a question here.

> which is often used to represent **a character** that has an alternative meaning when printed literally

**Escape characters** are like `&`, `%`, ... as mentioned below with examples. They are usually the preceding one(s) in **Escape sequences** like `&#x60;`, which in this example, it's actually 4/5chars (`#x60;`)?

---

```suggestion
An **escape character** is a {{glossary("character")}} that causes one or more characters that follow it to be interpreted differently. This forms an **escape sequence**, which is often used to signal that the following characters may have an alternative meaning when printed literally, such as the quote character in a string literal. Escape sequences can have other usage too, especially in [regular expressions](/en-US/docs/Web/JavaScript/Reference/Regular_expressions#escape_sequences).
```

Reviewer: What I mean is this: usually, escape characters are used in a context where almost all characters can be used to represent themselves. For example, `"abc"` means the characters a, b, and c. However, there are some characters that "has an alternative meaning when printed literally", for example you can't use `"""` to mean the quote character, because the quote itself is a syntax character. In this case, by escaping it, we actually "go back to the literal meaning". I'm not sure how to best phrase it, but what you proposed is not what I mean: "the following characters" is not necessarily the character that gets printed out. For example, the line break character is what gets printed out literally, but its escape sequence is `\n`, which does not contain the actual line break character.

Author: that's a point :)

> - In {{glossary("JavaScript")}}, we can use the backslash (`\`) to escape characters like `\'`, `\"`, `\u0026`, etc.

Reviewer: ```suggestion
- In {{glossary("JavaScript")}} [regexes](/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Character_escape), [string literals](/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#string_literals), and [identifiers](/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers), we can use the backslash (`\`) to escape characters like `\'`, `\"`, `\u0026`, etc.
```

> - In {{glossary("CSS")}}, we can use the backslash (`\`) to escape characters like `\\`, `\n`, `\26`, etc. See [escaping characters](/en-US/docs/Web/CSS/ident#escaping_characters) for more information.

Reviewer: ```suggestion
- In {{glossary("CSS")}} identifiers, we can use the backslash (`\`) to escape characters like `\\`, `\n`, `\26`, etc. See [escaping characters](/en-US/docs/Web/CSS/ident#escaping_characters) for more information.
```

> - In {{glossary("HTML")}}, we can use {{glossary("character reference", "character references")}} like `&lt;`, `&#60;`, or `&#x3C;` to represent the less-than (`<`) symbol.

Reviewer: ```suggestion
- In {{glossary("HTML")}} text content and attribute values, we can use {{glossary("character reference", "character references")}} like `&lt;`, `&#60;`, or `&#x3C;`.
```

> - In {{glossary("HTML")}}, we can use the {{glossary("character reference", "character references")}} like `&lt;`, `&#60;`, or `&#x3C;` to represent the less-than (`<`) symbol.

Reviewer 2: ```suggestion
- In {{glossary("HTML")}}, we can use {{glossary("character reference", "character references")}} like `&lt;`, `&#60;`, or `&#x3C;` to represent the less-than (`<`) symbol.
```

> - In {{glossary("URL")}}, we can use the percent sign (`%`) to escape characters like `%20`, `%3C`, `%3E`, etc.

Reviewer 2: ```suggestion
- In {{glossary("URL", "URLs")}}, we can use the percent sign (`%`) to escape characters like `%20`, `%3C`, `%3E`, etc.
```

## See also

> - [Escape character](https://en.wikipedia.org/wiki/Escape_character) on Wikipedia
> - [Escape sequence](https://en.wikipedia.org/wiki/Escape_sequence) on Wikipedia
> - Related glossary terms:
>   - {{glossary("Character")}}
>   - {{glossary("Character reference")}}
>   - {{glossary("Code point")}}

Reviewer 2: ```suggestion
- Related glossary terms:
  - {{glossary("Character")}}
  - {{glossary("Character reference")}}
  - {{glossary("Code point")}}
- [Escape character](https://en.wikipedia.org/wiki/Escape_character) on Wikipedia
- [Escape sequence](https://en.wikipedia.org/wiki/Escape_sequence) on Wikipedia
```
put internal links first.
